### ISSUE TITLE
[Machine learning] Implementing machine learning models using PyCaret

### ISSUE THREAD

**Issue by francis966** [section-engineering-education/engineering-education#7404]
#### Proposed title of article
[Machine learning] Implementing machine learning models using PyCaret
#### Proposed article introduction
PyCaret is an open-source, low-code machine learning library in Python that allows developers to go from preparing the data to deploying the model within a short period of time. PyCaret is essentially a Python wrapper around several machine learning libraries and frameworks such as [scikit-learn](https://scikit-learn.org/stable/), [XGBoost](https://xgboost.readthedocs.io/en/latest/), [LightGBM](https://github.com/microsoft/LightGBM), [CatBoost](https://catboost.ai/), [spaCy](https://spacy.io/), [Optuna](https://github.com/optuna/optuna), [Hyperopt](https://github.com/hyperopt/hyperopt) and [Ray](https://github.com/ray-project/ray/tree/master/python/ray/tune)

It is an end-to-end machine learning and model management tool that speeds up the experiment cycle exponentially and increases productivity. The goal of the caret package is to automate the major steps for evaluating and comparing machine learning algorithms for classification and regression. The main benefit of the library is that a lot can be achieved with very few lines of code and little manual configuration.

The design and simplicity of PyCaret are inspired by the emerging role of citizen data scientists, a term first used by Gartner. Citizen Data Scientists are power users who can perform both simple and moderately sophisticated analytical tasks that would previously have required more technical expertise. In this tutorial, we will process the heart disease dataset and use PyCaret to build a heart disease prediction model.

#### Key takeaways 
- Preparing heart disease dataset.
- Installing and Initializing the PyCaret setup
- Defining the data and the target class
- Compare multiple PyCaret models and select the best one.
- Evaluating the heart disease model and calculating the accuracy of the selected best model.
- Making predictions on unseen data.

#### Article quality
This article is unique because I will explain all the PyCaret components in detail and how they are used to automate the machine learning process. Also, we use PyCaret to run multiple machine learning models such as [scikit-learn](https://scikit-learn.org/stable/), [XGBoost](https://xgboost.readthedocs.io/en/latest/), [LightGBM](https://github.com/microsoft/LightGBM), [CatBoost](https://catboost.ai/), [spaCy](https://spacy.io/), [Optuna](https://github.com/optuna/optuna), [Hyperopt](https://github.com/hyperopt/hyperopt) and [Ray](https://github.com/ray-project/ray/tree/master/python/ray/tune). We will then compare the results of each model and only select the one with the best results.

I will also further fine-tune the selected model using a Grid Search technique and find the best model parameters. The final fine-tuned model will be fully optimized and we will apply it to unseen data to make predictions. The tutorial will be detailed with all the steps clearly explained so that anyone can easily follow and understand the tutorial.
#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

**Comment by hectorkambow** [comment]
Good afternoon and thank you for submitting your topic to the EngEd program. 

After some careful consideration it struck us that this topic may be a bit over saturated throughout other blog sites and official documentations. We typically refrain from publishing content that is covered widely on the net or other blogs. We're more interested in original, practitioner-focused content that takes a deeper dive into programming-centric concepts. 

But in order to approve the topic it has to serve value to the larger developer community at large. Please feel free to suggest an alternate topic to explore. 🚀 
An option and a great way to write this as an in-depth article and make it more add value to the greater developer community at large would be to walk the reader through the USE of methods and functions by building a unique, different, useful project. 

If you believe your article is unique enough - please present your case to explain how it will be different from what is currently out there?

https://www.google.com/search?q=machine+learning+models+using+PyCaret&rlz=1C1CHBF_enUS891US891&sourceid=chrome&ie=UTF-8
